src/ucp/core/ucp_mm.c

```diff
@@ -1014,6 +1022,13 @@ static ucs_rcache_ops_t ucp_mem_rcache_ops = {
 ucs_status_t ucp_mem_rcache_init(ucp_context_h context)
 {
     ucs_rcache_params_t rcache_params;
+    ucs_status_t status;
+    ucp_md_map_t dummy;
+
+    status = ucp_mm_get_alloc_md_map(context, &dummy);
+    if (status != UCS_OK) {
+        return status;
+    }
```

Comment:
why is this needed?

Reply:
There is an issue with cuMemAlloc - we need to call cuda register once to initialize cuda context.